src/components/CreateItem.jsx

```diff
@@ -1,18 +1,89 @@
-import React from 'react';
-import firebase from 'firebase/app';
+import React, { useState } from 'react';
+import db from '../lib/firebase';
+import './createItem.css';
 
 function CreateItem() {
-  const handleClick = async () => {
-    await firebase.firestore().collection('items').add({
-      name: 'A test product',
+  const itemObj = {
+    itemName: '',
+    frequency: 7,
+  };
+  const [item, setItem] = useState(itemObj);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const token = localStorage.getItem('token');
+
+    db.collection(token).add({
+      name: item.itemName,
+      frequency: item.frequency,
+      lastPurchasedDate: null,
       date: Date().toLocaleString(),
     });
```

Comment:
This works well now when we add tokens manually to our localStorage. However, this could be a very expensive transaction in terms of handling our database.

Currently, how this function _(logic)_ works is the following: 
When a new token is created, it also creates a new database collection in Firestore. And it is named after the token name. 

The way we want our database to work is to have one `collection` _(for example `items` as we had previously)_. Our collection should have many unique `documents` inside that can have one or many `objects` inside.

Therefore, I would suggest keeping the collection name `items`, the way you had previously: `db.collection('items')` and create the test token in a different way. 

There could be multiple ways to create a test token. Here is how I would do it:

```jsx
const handleSubmit = (e) => {
    e.preventDefault();
>> localStorage.setItem('token', 'example of token name'); <<

db.collection('items').add({
      name: item.itemName,
      frequency: item.frequency,
      lastPurchasedDate: null,
      date: Date().toLocaleString(),
    });
```
To explain the code: when we click `add item`, or when we submit the item to the db, the token is set to the localStorage with key: 'token' and value: 'example of token name'

I know that the other pair works on the token issue but I also wanted to explain why we need to keep one collection in the db and just create a simple `test token` the following way. 

Please let me know if anything is confusing because it could totally be me who is not as clear and detailed 💛

One more note: make sure to coordinate with the other pair to see how they set the token and how they add it to the database. It will be helpful for merging.
I think they decided to add a token as one of the properties of the `item object`, which is totally fine 👍

Reply:
Hi @Lukabaramishvili, so this was actually what we did originally, except we added the token as one of the properties of the added object above, like this:
```
db.collection('items').add({
      name: item.itemName,
      frequency: item.frequency,
      lastPurchasedDate: null,
      date: Date().toLocaleString(),
      token: token,
    });
```
However, when we had the office hours on Friday, Chiamaka said that we should only be displaying items relevant to the current user token, and this was the solution i proposed to her which she said looked ok. (our thread is in the slack channel)

The other pair have the job of creating and setting the token, and one of our ACs is to display the items as well as add them with an association to the user token.
If you still think there's an issue with the above, maybe it would be good to talk about it on a call today, if you and @ishimwezachee are free?

Thanks! 😁

Reply:
I would love it, if we  can talk about it  😁 😁 😁

Reply:
Adding to what @sponberry said, We also had a challenge of relating the data sent to its corresponding token, so that's how we came up with creating a collection based on its token.

Reply:
Yes, I saw the slack discussion. It is awesome that you thought about arranging the database so that each new token has an associated array of items inside. 

I think the right way to approach this solution is to have only 1 collection with many documents. These documents can be associated with (named after) new tokens. Finally, each document can be an array of objects that hold items associated with that specific token. 🙂

Your original implementation should work well. If you take the old route you will just need to check if items token property matches with the current browser localStorage token. And only display the items after that ✨

I am also happy to jump on a call, if you both want it 👍